src/sqlancer/stonedb/StoneDBSchema.java

```diff
@@ -0,0 +1,253 @@
+        case "varchar":
+            return new StoneDBCompositeDataType(StoneDBDataType.VARCHAR);
+        // todo: support more data type
+        default:
+            throw new AssertionError(typeString);
+        }
+    }
+
+
+    private static StoneDBDataType getColumnType(String typeString) {
+        switch (typeString) {
+        case "tinyint":
+            return StoneDBDataType.TINYINT;
+        case "smallint":
+            return StoneDBDataType.SMALLINT;
+        case "mediumint":
+            return StoneDBDataType.MEDIUMINT;
+        case "int":
+            return StoneDBDataType.INT;
+        case "bigint":
+            return StoneDBDataType.BIGINT;
+        case "float":
+            return StoneDBDataType.FLOAT;
+        case "double":
+            return StoneDBDataType.DOUBLE;
+        case "decimal":
+            return StoneDBDataType.DECIMAL;
+        case "char":
+            return StoneDBDataType.CHAR;
+        case "varchar":
+            return StoneDBDataType.VARCHAR;
+        // todo: support more data type
+        default:
+            throw new AssertionError(typeString);
+        }
+    }
+
+    private static List<String> getTableNames(SQLConnection con) throws SQLException {
+        List<String> tableNames = new ArrayList<>();
+        try (Statement s = con.createStatement()) {
+            try (ResultSet rs = s.executeQuery("SELECT * FROM sqlite_master WHERE type='table' or type='view'")) {
+                while (rs.next()) {
+                    tableNames.add(rs.getString("name"));
+                }
+            }
+        }
+        return tableNames;
+    }
+
+    public static class StoneDBColumn extends AbstractTableColumn<StoneDBTable, StoneDBCompositeDataType> {
+
+        private final boolean isPrimaryKey;
+        private final boolean isNullable;
+        private final int precision;
+
+        public StoneDBColumn(String name, StoneDBCompositeDataType columnType, boolean isPrimaryKey, boolean isNullable,
+                int precision) {
+            super(name, null, columnType);
+            this.isPrimaryKey = isPrimaryKey;
+            this.isNullable = isNullable;
+            this.precision = precision;
+        }
+
+        public boolean isPrimaryKey() {
+            return isPrimaryKey;
+        }
+
+        public boolean isNullable() {
+            return isNullable;
+        }
+
+    }
+
+    public StoneDBSchema(List<StoneDBTable> databaseTables) {
+        super(databaseTables);
+    }
+
+    public static class StoneDBCompositeDataType {
+        private final StoneDBDataType dataType;
+        private int size;
+
+        public StoneDBCompositeDataType(StoneDBDataType dataType, int size) {
+            this.dataType = dataType;
+            this.size = size;
+        }
+
+        public StoneDBCompositeDataType(StoneDBDataType dataType) {
+            this.dataType = dataType;
+            size = -1;
+            switch (dataType) {
+            case TINYINT:
```

Comment:
These data types are implemented differently in MySQL, but seem fine-grained and correct with reference to the doc: https://stonedb.io/docs/SQL-reference/data-types?_highlight=data .

Have you checked with the StoneDB doc? If so, it is good.